nidaqmx/_library_interpreter.py

```diff
@@ -0,0 +1,241 @@
+import ctypes
+from nidaqmx.errors import check_for_error
+
+from nidaqmx._lib import lib_importer, ctypes_byte_str
+
+class LibraryInterpreter(object):
+    '''Library C<->Python interpreter.
+
+    This class is responsible for interpreting the Library's C API. It is responsible for:
+    * Converting ctypes to native Python types.
+    * Dealing with string encoding.
+    * Allocating memory.
+    * Converting errors returned by Library into Python exceptions.
+    '''
+
+    def __init__(self, encoding):
+        self._encoding = encoding
+        self.set_task_handle()
+        self.set_system_handle()
+        self.set_scale_handle()
+    
+    def set_task_handle(self, value=0):
+        self._task_handle = value
+
+    def get_task_handle(self):
+        return self._task_handle
+    
+    def set_system_handle(self, value=0):
+        self._system_handle = value
+
+    def get_system_handle(self):
+        return self._system_handle
+
+    def set_scale_handle(self, value=0):
+        self._scale_handle = value
+
+    def get_scale_handle(self):
+        return self._scale_handle
+
+    def create_task(self, new_task_name):
+        
+        handle = lib_importer.task_handle(0)
+
+        cfunc = lib_importer.windll.DAQmxCreateTask
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        ctypes_byte_str,
+                        ctypes.POINTER(lib_importer.task_handle)]
+        error_code = cfunc(
+            new_task_name, ctypes.byref(handle))
+        check_for_error(error_code)
+
+        return handle
+    
+    def close(self):
+
+        cfunc = lib_importer.windll.DAQmxClearTask
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        lib_importer.task_handle]
+
+        error_code = cfunc(
+            self._task_handle)
+        check_for_error(error_code)
+    
+    def add_ai_voltage_chan(
```

Comment:
I think it would make more sense for the interpreter class method names to follow the C API, like `clear_task` and `create_ai_voltage_chan`.

Rationale: many of the Python method names, such as `close`, only make sense in the context of the class where they are defined.

Reply:
FYI, _grpc_interpreter.py has `clear_task` not `close`